pyugrid/read_netcdf.py

```diff
@@ -207,7 +207,7 @@ def load_grid_from_nc_dataset(nc, grid, mesh_name=None, load_data=True):
             except AttributeError:
                 start_index = 0
             if start_index >= 1:
-                array -= start_index
+                array -= start_index.astype(array.dtype)
```

Comment:
I'm a bit confused here: ``.astype`` is only going to work for a numpy array (or maybe a netcdf variable), but start_index came from: ``start_index = var.start_index`` -- i.e. a variable attribute -- wouldn't it be a string or scalar?

and += works for any integer dtype, as far as I can see:

```
In [7]: np.ones((3,), dtype = np.int16)
Out[7]: array([1, 1, 1], dtype=int16)

In [8]: arr = np.ones((3,), dtype = np.int16)

In [9]: arr += 2

In [10]: arr
Out[10]: array([3, 3, 3], dtype=int16)

In [11]: arr += np.int32(4)

In [12]: arr
Out[12]: array([7, 7, 7], dtype=int16)

In [13]: arr += np.int64(4)

In [14]: arr
Out[14]: array([11, 11, 11], dtype=int16)
```